### PR TITLE
feat(DatePicker): Today button

### PR DESCRIPTION
**Pull request by jillyan**
**What is the problem this PR is trying to solve?**
TCCUI-65 (Date Picker Improvement) Add a Today action.
**What is the chosen solution to this problem?**
Add Today action in the date and year/month view, when click Today button, it will select the current year-month-day.
**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss [hunk 1]
```diff
@@ -7,3 +9,33 @@
...
+}
+
+.btn-today {
+	color: $st-tropaz;
+	font-weight: bold;
+	text-transform: uppercase;
+	padding: 0;
+	height: 25px;
+	> span {
+		border-radius: 4px;
```
Comment (frassinier) [inline packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss]: there is a variable theme for that I guess, look for 'border-radius'

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss [hunk 2]
```diff
@@ -7,3 +9,33 @@
...
+		}
+	}
+}
+
+.btn-today {
+	color: $st-tropaz;
+	font-weight: bold;
+	text-transform: uppercase;
+	padding: 0;
+	height: 25px;
```
Comment (frassinier) [inline packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss]: Use rem unit please

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.test.js [hunk 3]
```diff
@@ -227,4 +227,21 @@ describe('DateTimePicker', () => {
...
+			wrapper.setState({ isDateTimeView: false });
+
+			// when
+			wrapper
+				.find('.theme-btn-today')
+				.at(0)
+				.simulate('click');
+
+			// then
+			expect(wrapper.state('isDateTimeView')).toBe(true);
```
Comment (frassinier) [inline packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.test.js]: You should test that "today" is the new selected date

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.component.js [hunk 4]
```diff
@@ -173,6 +187,19 @@ class DateTimePicker extends React.Component {
 				aria-label="Date picker"
 			>
 				{viewElement}
+				<div className={theme.footer}>
+					<Action
+						label={this.props.t('DATEPICKER_TODAY_ACTION', {
+							defaultValue: 'Today',
+						})}
+						aria-label={this.props.t('DATEPICKER_SELECT_TODAY', {
```
Comment (frassinier) [inline packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.component.js]: Nice one :+1: but you can improve it by describing a few more _Pick today as selected date_ or something like that

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss [hunk 5]
```diff
@@ -7,3 +9,33 @@
...
+
+.footer {
+	:global {
+		.btn-link {
+			color: $st-tropaz;
+			text-decoration: none;
+
+			:hover {
+				background-color: rgba($st-tropaz, 0.1);
+				transition: opacity 0.1s ease-in;
```
[the remark this answers is not in the document]
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss]: Hi @frassinier ,
Thanks for reviewing :).
I set it in rgba($st-tropaz, 0.1) and the opacity is 0.1

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss [hunk 6]
```diff
@@ -7,3 +9,33 @@
...
+			}
+			:focus {
+				background-color: rgba($st-tropaz, 0.2);
+			}
+		}
+	}
+}
+
+.btn-today {
+	color: $st-tropaz;
```
[the remark this answers is not in the document]
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss]: thanks, updated already. :-)

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss (hunk 1 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss]: thanks, updated already. :-)

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss (hunk 2 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss]: thanks, updated already. :-)

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.test.js (hunk 3 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.test.js]: thanks, updated already. :-)

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.component.js (hunk 4 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.component.js]: Sorry, I don't get you.
Do you mean set today as default?

Comment (frassinier) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.component.js]: I suggest you improve the aria-label value by choosing _Pick today as selected date_ instead

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (frassinier) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss]: So transition should concern `background-color`: `transition: background-color 0.1s ease-in;`
And this transition property should be placed on default state, not on the `hover` or `focus`

**Comment by mgirault** [comment]
My remarks : 
1) [HTML tag] btn-today is not linked to the datePicker (DateTimePicker__btn-today), it’s single component like btn-secondary, we decided to call him “btn-tertiary. We should have added it in the specification sorry
2) [Style] the height is not 3.5rem (35px) it’s 2.5rem (25px)
3) [Style] the font weight is not 700 it’s 600, sorry not mentionned in the button spec
4) [Style] I didn’t see the transition (background 0.2s ease-in)
5) [Style] It’s not rgba for the hover effect it’s this function : tint ($st-tropaz, 90%) 
6) [Style] I didn’t see the pressed (active) effect, tint ($st-tropaz, 80%)
7) [Global effect] I didn’t see style http://guidelines.talend.com/document/92132#/design-principles/accessibility

File: output/components.eslint.txt [hunk 7]
```diff
@@ -19,6 +19,9 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   33:11  error  Caution: `i18next` also has a named export `t`. Check if you meant to write `import {t} from 'i18next'` instead  import/no-named-as-default-member
   35:11  error  Caution: `i18next` also has a named export `t`. Check if you meant to write `import {t} from 'i18next'` instead  import/no-named-as-default-member
 
+/home/travis/build/Talend/ui/packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.component.js
+  120:2  error  setAllowFocus should be placed after onClickToday  react/sort-comp
```
Comment (jsomsanith-tlnd) [inline output/components.eslint.txt]: Please fix this lint error

File: packages/components/src/DateTimePickers/DateTime/Picker/Picker.component.test.js [hunk 8]
```diff
@@ -25,6 +25,7 @@ describe('DateTime.Picker', () => {
 				useSeconds: true,
 				useUTC: false,
 			},
+			t: jest.fn(),
```
Comment (jsomsanith-tlnd) [inline packages/components/src/DateTimePickers/DateTime/Picker/Picker.component.test.js]: Why did you add that in the test ?

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.component.js [hunk 9]
```diff
@@ -173,6 +187,18 @@ class DateTimePicker extends React.Component {
 				aria-label="Date picker"
 			>
 				{viewElement}
+				<div>
+					<Action
+						label={this.props.t('DATEPICKER_TODAY_ACTION', {
```
Comment (jsomsanith-tlnd) [inline packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.component.js]:
```suggestion
						label={this.props.t('DATEPICKER_TODAY', {
```

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss]: the css is moved to tertiary button, theme package.

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss]: css is moved to tertiary button, theme package

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.test.js [hunk 10]
```diff
@@ -227,4 +227,22 @@ describe('DateTimePicker', () => {
 			expect(onSubmit).toBeCalledWith(event, { date: initialDate, time });
 		});
 	});
+
+	describe('today function', () => {
```
Comment (jsomsanith-tlnd) [inline packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.test.js]: Can you add a test that check that the picked value is selected too ?
You can look at the other tests to have examples

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss (hunk 1 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss]: css is moved to tertiary button, theme package

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.test.js (hunk 10 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.test.js]: I have this 
`expect(wrapper.state('selectedDate')).toStrictEqual(startOfDay(new Date()));`

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss (hunk 2 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.scss]: style is moved to tertiary button, theme package.

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.test.js (hunk 10 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.test.js]: There is 
`expect(wrapper.state('selectedDate')).toStrictEqual(startOfDay(new Date()));`

File: packages/components/src/DateTimePickers/DateTime/Picker/Picker.component.test.js (hunk 8 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/DateTime/Picker/Picker.component.test.js]: removed it.

File: packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.component.js (hunk 9 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DateTimePicker/DateTimePicker.component.js]: updated

File: packages/components/src/DateTimePickers/DateTime/Picker/Picker.component.test.js [hunk 11]
```diff
@@ -46,6 +46,7 @@ describe('DateTime.Picker', () => {
 			useSeconds: true,
 			useTime: true,
 			useUTC: false,
+			t: expect.any(Function),
```
Comment (jsomsanith-tlnd) [inline packages/components/src/DateTimePickers/DateTime/Picker/Picker.component.test.js]: You forgot to remove that

Comment (jillyan) [reply packages/components/src/DateTimePickers/DateTime/Picker/Picker.component.test.js]: this is expected to be here, since there is function `t` in the DateTimePicker
